### ISSUE TITLE
HMI SoC is inaccurate on 45/55kwh ID.3

### ISSUE THREAD

**Issue by tolnaiz** [nickn17/evDash#50]
This calculation seems to be inaccurate on the pure models.
https://github.com/nickn17/evDash/blob/ad4c1522bd5018f0b21e61c7bbc5d09af14db81c/src/CarVWID3.cpp#L315

I think it would be much more reliable to use "battery energy content" divided by "battery max energy content" formula here.
I guess this calculation would be accurate on all models.

Also, batModuleTempCount is 16 and cellCount is 96 on the pure model.
Basically vw just removed one battery module from the 58/62kwh version.

**Comment by tolnaiz** [comment]
The biggest differences are above 50%.

This is what I see in the car with OBD:
Car reports 100%, battery BMS SoC is at 86,4% with EvDash new formula it's 89,38%
At 76%, BMS is at 69,2% when evDash would be: 70,3
At 51%, BMS is at 50%, evDash: 49,11%
At 18%, BMS is at 22,8%, evDash: 19,02%
The lowest SoC I was able to record with OBD is 11% when the BMS SoC was at 16%, evDash: 11,5

Unfortunately these numbers doesn't even fit on a slope:
https://www.wolframalpha.com/input?i=plot+points%5B%2816%2C+11%29%2C+%2822.8%2C+18%29%2C+%2850%2C+51%29%2C+%2869.2%2C+76%29%2C+%2886.4%2C+100%29%5D

I measured these numbers at very different time, so maybe temperature, SoH or other factors could affect the calculation.

The best formula I was able to figure out is this: y = x * 0,505-9,0947
This fits better, but still not perfect.
